src/test/common/variables/kernelEnvVarsService.unit.test.ts

```diff
@@ -0,0 +1,119 @@
+        envType: EnvironmentType.Conda,
+        path: 'foobar',
+        sysPrefix: '0'
+    };
+    const kernelSpec: IJupyterKernelSpec = {
+        name: 'kernel',
+        path: 'foobar',
+        display_name: 'kernel',
+        interpreterPath: 'foobar',
+        argv: []
+    };
+
+    setup(() => {
+        fs = TypeMoq.Mock.ofType<IFileSystem>(undefined, TypeMoq.MockBehavior.Strict);
+        envActivation = TypeMoq.Mock.ofType<IEnvironmentActivationService>(undefined, TypeMoq.MockBehavior.Strict);
+        customVariablesService = TypeMoq.Mock.ofType<IEnvironmentVariablesProvider>(
+            undefined,
+            TypeMoq.MockBehavior.Strict
+        );
+        interpreterService = TypeMoq.Mock.ofType<IInterpreterService>(undefined, TypeMoq.MockBehavior.Strict);
+        variablesService = new EnvironmentVariablesService(fs.object);
+        kernelVariablesService = new KernelEnvironmentVariablesService(
+            interpreterService.object,
+            envActivation.object,
+            variablesService,
+            customVariablesService.object
+        );
+    });
+
+    suite(`getEnvironmentVariables()`, () => {
+        test('Interpreter path trumps process', async () => {
+            envActivation
+                .setup((e) =>
+                    e.getActivatedEnvironmentVariables(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())
+                )
+                .returns(() => Promise.resolve({ PATH: 'foobar' }));
+            customVariablesService
+                .setup((c) => c.getCustomEnvironmentVariables(TypeMoq.It.isAny()))
+                .returns(() => Promise.resolve(undefined));
+
+            const vars = await kernelVariablesService.getEnvironmentVariables(undefined, interpreter, kernelSpec);
+
+            const processPath = Object.keys(process.env).find((k) => k.toLowerCase() == 'path');
+            expect(processPath).to.not.be.undefined;
+            expect(vars).to.not.be.undefined;
+            expect(vars![processPath!]).to.be.equal('foobar');
+        });
+
+        test('Paths are merged', async () => {
+            envActivation
+                .setup((e) =>
+                    e.getActivatedEnvironmentVariables(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())
+                )
+                .returns(() => Promise.resolve({ pATh: 'foobar' }));
+            customVariablesService
+                .setup((c) => c.getCustomEnvironmentVariables(TypeMoq.It.isAny()))
+                .returns(() => Promise.resolve({ PATH: 'foobaz' }));
+
+            const vars = await kernelVariablesService.getEnvironmentVariables(undefined, interpreter, kernelSpec);
+            const processPath = Object.keys(process.env).find((k) => k.toLowerCase() == 'path');
+            expect(processPath).to.not.be.undefined;
+            expect(vars).to.not.be.undefined;
+            expect(vars![processPath!]).to.be.equal(`foobar${path.delimiter}foobaz`);
```

Comment:
Fix here in this update to use path.delimiter as test was failing on my machine due to : versus ;.